src/d3a/d3a_core/live_events.py

```diff
@@ -46,11 +49,31 @@ def __init__(self, area_uuid, area_params):
     def apply(self, area):
         if area.uuid != self.area_uuid:
             return False
-
-        area.area_reconfigure_event(**self.area_params)
+        self.sanitize_live_event_paramters_for_strategy_update()
+        if key_in_dict_and_not_none(self.area_params, 'type'):
+            if area.strategy is None:
```

Comment:
you could have a clause `area_type = self.area_params.pop("type")`. That way you will avoid the explicit del's below